### DISCUSSION TITLE
(mini.pick) - how to pick files from specific directory?

### DISCUSSION THREAD

**Discussion by mainendra** [nvim-mini/mini.nvim#642] [Q&A] [answered]
`Pick files` will always show list of files in cwd. How to pick files from different directory without changing cwd.

**Comment by echasnovski** [comment, marked as answer]
Every source can have "current working directory". So with `builtin.files()` you can do `MiniPick.builtin.files(nil, { source = { cwd = '../foo/bar' } })`.

If you plan to do that often interactively, you can [modify `files` in registry](https://github.com/echasnovski/mini.nvim/blob/a118a964c94543c06d8b1f2f7542535dd2e19d36/doc/mini-pick.txt#L1186-L1191) and use `:Pick files cwd='../foo/bar'`.

**Comment by shubham-cpp** [comment]
@echasnovski, having one with the code you've provided. If I hit enter on the selection(i.e "choose" the selection)  then it selects the file, but if I "Ctrl+v"(i.e "choose_in_vsplit") then it changes my directory.

<details>

<summary>My `mini.pick` config</summary>

```lua
local function pick_cmd(cmd)
  return "<cmd>Pick " .. cmd .. "<cr>"
end
---@type LazySpec
return {
  'echasnovski/mini.pick',
  dependencies = {
    { 'echasnovski/mini.icons', config = function() require('mini.icons').setup() end },
    {
      'echasnovski/mini.extra',
      config = function()
        require('mini.extra').setup()
      end,
    },
  },
  cmd = { 'Pick' },
  keys = {
    { '<C-p>',     pick_cmd("files"), desc = '[F]ile' },
    {
      '<leader>fn',
      pick_cmd(string.format("files cwd='%s'", vim.fn.stdpath("config"))),
      desc = '[N]eovim config',
    },
  }
  opts = {
    mappings = {
      choose_in_split = '<C-x>',
      mark = '<C-l>',
      move_down = '<C-j>',
      move_up = '<C-k>',
    }
    options = { use_cache = true },
  },
  config = function(_, opts)
    local picker = require("mini.pick")
    picker.setup(opts)
    -- Make `:Pick files` accept `cwd`
    MiniPick.registry.files = function(local_opts)
      local opts = { source = { cwd = local_opts.cwd } }
      local_opts.cwd = nil
      return MiniPick.builtin.files(local_opts, opts)
    end
    vim.ui.select = picker.ui_select
  end
}

```

</details>


https://github.com/user-attachments/assets/cc7b0a42-7a92-41bf-85d1-426c00cefe6c

**Reply by echasnovski** [reply]
Yes, I can reproduce. This is probably a side effect of https://github.com/echasnovski/mini.nvim/commit/5833f01484ad81da31d175e53468f761994daeaa , which should not happen. I'll take a look.

**Reply by shubham-cpp** [reply]
This looks like a rabbit hole. 😅
(Thank you for quick response 🙏🏼)

Maybe I'll check the code from `snacks.picker` to know how it handles this situation.
